.github/workflows/update-submodules.yml

```
@@ -4,6 +4,12 @@ on:
   schedule:
     - cron: "0 8 * * *"
 jobs:
+  functional-tests:
+    uses: ./.github/workflows/functional.yml
+    with:
+      api_version: dev
+      worker_version: dev
+      cli_version: dev
   sync-submodules:
     name: "Sync submodules"
     runs-on: ubuntu-latest
```

We need to specify that the sync-submodules depend on `functional-tests`; otherwise, it will open the PR even if the FT fails.

```suggestion
    runs-on: ubuntu-latest
    needs: functional-tests
```

---

Thanks, after so many edits on this file totally missed that out 😄